### PR TITLE
Add highlighting to search API

### PR DESCRIPTION
**Pull request by tiramisu24**
-searchMethod tests
-change specs to have highlight field and option to turn it on/off

### CODE REVIEW

File: lib/src/kbasesearchengine/main/SearchMethods.java [hunk 1]
```diff
@@ -183,11 +183,14 @@ private Pagination fromSearch(kbasesearchengine.search.Pagination pg) {
...
             ret.objectData = true;
             ret.objectKeys = true;
+            ret.objectHighlight = false;
         } else {
             boolean idsOnly = toBool(pp.getIdsOnly());
             ret.objectInfo = !(toBool(pp.getSkipInfo()) || idsOnly);
             ret.objectData = !(toBool(pp.getSkipData()) || idsOnly);
             ret.objectKeys = !(toBool(pp.getSkipKeys()) || idsOnly);
+            //default to false currently b/c of search tags. TODO: add search tags to black list?
+            ret.objectHighlight = toBool(pp.getIncludeHighlight()) && !idsOnly;
```
Comment (MrCreosote) [inline lib/src/kbasesearchengine/main/SearchMethods.java]: This line is only partially covered

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 2]
```diff
@@ -277,5 +294,66 @@ private void searchTypesCheckMatchFilter(
...
+
+        final SearchObjectsOutput res = sm.searchObjects(new SearchObjectsInput()
+                        .withMatchFilter(new MatchFilter().withFullTextInAll("test"))
+                        .withAccessFilter(new AccessFilter()), "auser");
+
+
+        final Map<String, List<String>> highlightRes = new HashMap<>();
+        highlightRes.put("field", highlight);
+
+        assertThat("did not find objects", res.getObjects().size()>0, is(true));
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: spaces around operators

Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: Why not just size == 1?

File: test/src/kbasesearchengine/test/search/ElasticIndexingStorageTest.java [hunk 3]
```diff
@@ -1124,7 +1123,18 @@ public void addHighlighting() throws Exception {
         final Builder filter = MatchFilter.getBuilder();
         filter.withNullableFullTextInAll("multiWordInSearchMethod1 multiWordInSearchMethod2");
 
+
+        //highlight turned off would give null objects unless objectData is set to true
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/search/ElasticIndexingStorageTest.java]: Can you note that the purpose of this test is to show that turning highlighting off works?

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 4]
```diff
@@ -277,5 +294,66 @@ private void searchTypesCheckMatchFilter(
         // if we want to check search time, need to mock a Clock. Don't bother for now.
         // assertThat("incorrect objects", res.getSearchTime(), is(20));
     }
-    
+
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: empty lines

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 5]
```diff
@@ -277,5 +294,66 @@ private void searchTypesCheckMatchFilter(
...
+
+
+
+    @Test
+    public void searchObjectResult() throws Exception {
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: There's at least 4 different cases that need to be covered in SearchMethods diff - I think this only covers one?

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 6]
```diff
@@ -277,5 +294,66 @@ private void searchTypesCheckMatchFilter(
...
+        pp.objectHighlight = true;
+
+
+        final SortingRule sr = new SortingRule();
+        sr.isTimestamp = true;
+        sr.ascending = true;
+
+        final FoundHits fh = new FoundHits();
+        fh.pagination = null;
+        fh.sortingRules=  Arrays.asList(sr);;
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: spaces

File: lib/src/kbasesearchengine/search/ElasticIndexingStorage.java [hunk 7]
```diff
@@ -75,6 +75,24 @@
     private static final String SEARCH_OBJ_TYPE = "otype";
     private static final String SEARCH_OBJ_TYPE_VER = "otypever";
 
+    //skipped timestamp, md5, copier, creator
```
Comment (MrCreosote) [inline lib/src/kbasesearchengine/search/ElasticIndexingStorage.java]: Just be explicit, even if there's no changes throw em in there for consistency

File: lib/src/kbasesearchengine/main/SearchMethods.java (hunk 1 above)
Comment (tiramisu24) [reply lib/src/kbasesearchengine/main/SearchMethods.java]: you mean also check with idsOnly?

Comment (MrCreosote) [reply lib/src/kbasesearchengine/main/SearchMethods.java]: Dunno. Whatever getting full coverage of the line takes.

File: test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java [hunk 8]
```diff
@@ -342,6 +345,10 @@ public void sourceTags() throws Exception {
         
         assertThat("incorrect object count", res2.getObjects().size(), is(1));
         compare(res2.getObjects().get(0), expected1);
+
```
[the remark this answers is not in the document]
Comment (tiramisu24) [reply test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java]: no idea

File: lib/src/kbasesearchengine/search/ElasticIndexingStorage.java [hunk 9]
```diff
@@ -75,6 +75,24 @@
...
+    private static final ImmutableMap<String, String> readableNames = ImmutableMap.<String,String>builder()
+            .put(OBJ_PROV_COMMIT_HASH, "provenance_commit")
+            .put(OBJ_PROV_MODULE_VERSION, "provenance_module_ver")
+            .put(OBJ_PROV_METHOD, "provenance_method")
+            .put(OBJ_PROV_MODULE, "provenance_module")
+            .put(OBJ_NAME, "object_name")
+            .put(SOURCE_TAGS, "source_tags")
+            .put(SEARCH_OBJ_TYPE, "type")
+            .put(SEARCH_OBJ_TYPE_VER, "type_ver")
+            .put("str_cde", "storage_code")
```
[the remark this answers is not in the document]
Comment (tiramisu24) [reply lib/src/kbasesearchengine/search/ElasticIndexingStorage.java]: okay. I miss javascript

File: lib/src/kbasesearchengine/search/ElasticIndexingStorage.java [hunk 10]
```diff
@@ -1162,22 +1180,27 @@ private ObjectData buildObjectData(
...
     }
 
+    private String getReadableKeyNames(final String key){
+        if (key.startsWith("key.")) {
+            return stripKeyPrefix(key);
+        }
+        if(readableNames.containsKey(key)){
+            return readableNames.get(key);
+        }
+        return key;
```
[the remark this answers is not in the document]
Comment (MrCreosote) [reply lib/src/kbasesearchengine/search/ElasticIndexingStorage.java]: They should, what else is there?

File: lib/src/kbasesearchengine/search/ElasticIndexingStorage.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (MrCreosote) [reply lib/src/kbasesearchengine/search/ElasticIndexingStorage.java]: No, no you don't

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java (hunk 5 above)
Comment (tiramisu24) [reply test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: there is getObjectsByIds. <= I can add this in the rare case someone desperately wants the guids back highlighted

        //searchIds is a wrapper around queryHits and does not return object data and so will not be highlighted
        //searchTypes returns the number of items per type that. Highlight is not added b/c does not call buildObjects

File: test/src/kbasesearchengine/test/search/ElasticIndexingStorageTest.java (hunk 3 above)
Comment (tiramisu24) [reply test/src/kbasesearchengine/test/search/ElasticIndexingStorageTest.java]: okay

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java (hunk 5 above)
Comment (MrCreosote) [reply test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: We talked this one out, was talking about this line

`ret.objectHighlight = toBool(pp.getIncludeHighlight()) && !idsOnly;`

File: lib/src/kbasesearchengine/search/ElasticIndexingStorage.java (hunk 10 above)
[the remark this answers is not in the document]
Comment (tiramisu24) [reply lib/src/kbasesearchengine/search/ElasticIndexingStorage.java]: dunno. there are a bunch of fields that are not constants sprinkled in the code. Is "extpub" a field? its not in fns convertObject or buildObjectData

[the remark this answers is not in the document]
Comment (MrCreosote) [reply lib/src/kbasesearchengine/search/ElasticIndexingStorage.java]: If it's not in either of those functions then it's not an object property, right?

[the remark this answers is not in the document]
Comment (tiramisu24) [reply lib/src/kbasesearchengine/search/ElasticIndexingStorage.java]: also pguid and parent fields?

[the remark this answers is not in the document]
Comment (MrCreosote) [reply lib/src/kbasesearchengine/search/ElasticIndexingStorage.java]: What about them?

File: KBaseSearchEngine.spec [hunk 11]
```diff
@@ -163,7 +166,10 @@ module KBaseSearchEngine {
     /*
       Properties of found object including metadata, raw data and
           keywords.
-          
+      mapping<string, list<string>> highlight - The keys are the field names and the list 
+          contains the sections in each field that matched the search query. Fields with no
+          hits will not be available. Short fields that matched are shown in their entirety.
+          Longer fields are shown as snippets precedded or followed by "...".
```
Comment (MrCreosote) [inline KBaseSearchEngine.spec]: preceded

File: lib/src/kbasesearchengine/main/SearchMethods.java [hunk 12]
```diff
@@ -183,11 +183,15 @@ private Pagination fromSearch(kbasesearchengine.search.Pagination pg) {
...
             ret.objectData = true;
             ret.objectKeys = true;
+            ret.objectHighlight = false;
         } else {
             boolean idsOnly = toBool(pp.getIdsOnly());
             ret.objectInfo = !(toBool(pp.getSkipInfo()) || idsOnly);
             ret.objectData = !(toBool(pp.getSkipData()) || idsOnly);
             ret.objectKeys = !(toBool(pp.getSkipKeys()) || idsOnly);
+            //default to false currently b/c of search tags. TODO: add search tags to black list?
+            boolean test =  toBool(pp.getIncludeHighlight());
```
Comment (MrCreosote) [inline lib/src/kbasesearchengine/main/SearchMethods.java]: This variable looks unused

File: lib/src/kbasesearchengine/search/ElasticIndexingStorage.java [hunk 13]
```diff
@@ -69,12 +70,71 @@
...
+    private static final String R_OBJ_SHARED = "is_shared";
+
+    // tags on the data originating at the source of the data
+    private static final String R_SOURCE_TAGS = "source_tags";
+
+    private static final String R_SEARCH_OBJ_TYPE = "type";
+    private static final String R_SEARCH_OBJ_TYPE_VER = "type_ver";
+
+
+    //skipped timestamp, md5, copier, creator
```
Comment (MrCreosote) [inline lib/src/kbasesearchengine/search/ElasticIndexingStorage.java]: no longer true

File: lib/src/kbasesearchengine/search/ElasticIndexingStorage.java [hunk 14]
```diff
@@ -69,12 +70,71 @@
...
+    private static final String R_OBJ_TIMESTAMP = "timestamp";
+    private static final String R_OBJ_PROV_COMMIT_HASH = "provenance_commit";
+    private static final String R_OBJ_PROV_MODULE_VERSION = "provenance_module_ver";
+    private static final String R_OBJ_PROV_METHOD = "provenance_method";
+    private static final String R_OBJ_PROV_MODULE = "provenance_module";
+    private static final String R_OBJ_MD5 = "md5";
+    private static final String R_OBJ_COPIER = "copier";
+    private static final String R_OBJ_CREATOR = "creator";
+    private static final String R_OBJ_NAME = "object_name";
+    private static final String R_OBJ_PREFIX = "prefix";
```
Comment (MrCreosote) [inline lib/src/kbasesearchengine/search/ElasticIndexingStorage.java]: guid_prefix

File: lib/src/kbasesearchengine/search/ElasticIndexingStorage.java [hunk 15]
```diff
@@ -1162,22 +1222,26 @@ private ObjectData buildObjectData(
...
         }
 
         return b.build();
     }
 
+    private String getReadableKeyNames(final String key){
+        if (key.startsWith("key.")) {
+            return stripKeyPrefix(key);
+        }
+        if(readableNames.containsKey(key)){
```
Comment (MrCreosote) [inline lib/src/kbasesearchengine/search/ElasticIndexingStorage.java]: spaces

File: lib/src/kbasesearchengine/search/ElasticIndexingStorage.java [hunk 16]
```diff
@@ -1162,22 +1222,26 @@ private ObjectData buildObjectData(
...
+            return readableNames.get(key);
+        }
+        return key;
```
Comment (MrCreosote) [inline lib/src/kbasesearchengine/search/ElasticIndexingStorage.java]: If it's not a key property and not one of the known names, that seems like a bug and an exception should be thrown, right?

File: test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java (hunk 8 above)
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java]: remove excess blank lines

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 17]
```diff
@@ -277,5 +293,138 @@ private void searchTypesCheckMatchFilter(
         // if we want to check search time, need to mock a Clock. Don't bother for now.
         // assertThat("incorrect objects", res.getSearchTime(), is(20));
     }
-    
+
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: empty lines

File: test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java [hunk 18]
```diff
@@ -401,6 +467,8 @@ private SearchObjectsOutput searchObjects(final MatchFilter mf) throws Exception
         }
     }
 
+
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java]: extra lines

File: test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java [hunk 19]
```diff
@@ -390,6 +397,65 @@ public void narrativeDecoration() throws Exception {
...
+                        .build(),
+                Instant.ofEpochMilli(10000),
+                null,
+                new GUID("WS:1/1/1"),
+                ImmutableMap.of(new GUID("WS:1/1/1"), new ParsedObject(
+                        "{\"whee\": \"imaprettypony1\"}",
+                        ImmutableMap.of("whee", Arrays.asList("imaprettypony1")))),
+                false);
+
+        final SearchObjectsOutput res1 = searchObjects(new MatchFilter()
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java]: this variable looks unused

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 20]
```diff
@@ -28,16 +30,30 @@
 import kbasesearchengine.AccessFilter;
 import kbasesearchengine.Pagination;
 import kbasesearchengine.test.common.TestCommon;
+import org.junit.Test;
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: duplicate import

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 21]
```diff
@@ -28,16 +30,30 @@
...
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static kbasesearchengine.test.common.TestCommon.set;
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: duplicate import

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 22]
```diff
@@ -28,16 +30,30 @@
 import kbasesearchengine.AccessFilter;
 import kbasesearchengine.Pagination;
 import kbasesearchengine.test.common.TestCommon;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: Duplicate import

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 23]
```diff
@@ -28,16 +30,30 @@
...
+import java.util.Collections;
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: Duplicate import

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 24]
```diff
@@ -277,5 +293,138 @@ private void searchTypesCheckMatchFilter(
...
+        final PostProcessing pp2 = new PostProcessing();
+        pp2.objectHighlight = false;
+        pp2.objectData = false;
+        pp2.objectKeys = false;
+        pp2.objectInfo = false;
+        pp2.objectDataIncludes=null;
+
+        final FoundHits fh1 = new FoundHits();
+        fh1.pagination = null;
+        fh1.sortingRules =  Arrays.asList(sr);;
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: extra ;

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 25]
```diff
@@ -277,5 +293,138 @@ private void searchTypesCheckMatchFilter(
...
+        fh1.total = 1;
+        fh1.guids = set(new GUID("ws:1/2/3"));
+        fh1.objects = objs;
+
+        final FoundHits fh2 = new FoundHits();
+        fh2.pagination = null;
+        fh2.sortingRules =  Arrays.asList(sr);;
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: extra ;

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 26]
```diff
@@ -277,5 +293,138 @@ private void searchTypesCheckMatchFilter(
...
+                .withIdsOnly(1L);
+
+        final SearchObjectsOutput res3 = searchObjects(idx, option3);
+
+        assertThat("did not find objects", res3.getObjects().size() == 1, is(true));
+        assertThat("did not get right highlight", res3.getObjects().get(0).getHighlight(), is(Collections.emptyMap()));
+
+    }
+
+    private SearchObjectsOutput searchObjects(IndexingStorage idx, kbasesearchengine.PostProcessing pp) throws Exception{
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: space, final args

File: KBaseSearchEngine.spec [hunk 27]
```diff
@@ -130,13 +130,16 @@ module KBaseSearchEngine {
           ('key_props' field in ObjectData structure),
       skip_data - do not include raw data for object ('data' and 
           'parent_data' fields in ObjectData structure),
+      include_highlight - include highlights of fields that
+           matched query,
       ids_only - shortcut to mark all three skips as true.
```
Comment (MrCreosote) [inline KBaseSearchEngine.spec]: ...and highlight as false.

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 28]
```diff
@@ -277,5 +293,138 @@ private void searchTypesCheckMatchFilter(
         // if we want to check search time, need to mock a Clock. Don't bother for now.
         // assertThat("incorrect objects", res.getSearchTime(), is(20));
     }
-    
+
+
+
+
+    @Test
+    public void searchObjectResult() throws Exception {
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: For future reference this should really be split up into separate tests. It's crazy long and the various cases could be a test each.

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 29]
```diff
@@ -28,16 +30,30 @@
...
-    
+
     private final static PostProcessing PP_DEFAULT = new PostProcessing();
     static {
         PP_DEFAULT.objectData = true;
         PP_DEFAULT.objectInfo = true;
         PP_DEFAULT.objectKeys = true;
     }
 
+
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: extra lines

File: test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java [hunk 30]
```diff
@@ -14,8 +14,11 @@
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
+import com.fasterxml.jackson.databind.deser.DataFormatReaders;
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java]: Is this used?

File: KBaseSearchEngine.spec (hunk 27 above)
Comment (tiramisu24) [reply KBaseSearchEngine.spec]: ??

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java (hunk 28 above)
Comment (tiramisu24) [reply test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: its long mainly b/c of set up. how would you suggest splitting this up?

File: test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java (hunk 30 above)
Comment (tiramisu24) [reply test/src/kbasesearchengine/test/integration/SearchAPIIntegrationTest.java]: no. this probably got auto imported when I typoed.

File: KBaseSearchEngine.spec (hunk 27 above)
Comment (MrCreosote) [reply KBaseSearchEngine.spec]: ids_only description is no longer accurate.

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java (hunk 28 above)
Comment (MrCreosote) [reply test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: Talked about this in person

File: lib/src/kbasesearchengine/search/ElasticIndexingStorage.java [hunk 31]
```diff
@@ -1162,22 +1220,27 @@ private ObjectData buildObjectData(
...
     }
 
+    private String getReadableKeyNames(final String key) throws IllegalStateException{
+        if (key.startsWith("key.")) {
+            return stripKeyPrefix(key);
+        } else if(readableNames.containsKey(key)) {
+            return readableNames.get(key);
+        } else {
+            //this should not happen. Untested
+            throw new IllegalStateException("Object has unexpected key");
```
Comment (MrCreosote) [inline lib/src/kbasesearchengine/search/ElasticIndexingStorage.java]: Include the offending key in the error message. It'd be nice if it included the guid as well but we can leave that for later

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 32]
```diff
@@ -277,5 +285,136 @@ private void searchTypesCheckMatchFilter(
...
+                .withIdsOnly(1L);
+
+        final SearchObjectsOutput res3 = searchObjects(idx, option3);
+
+        assertThat("did not find objects", res3.getObjects().size() == 1, is(true));
+        assertThat("did not get right highlight", res3.getObjects().get(0).getHighlight(), is(Collections.emptyMap()));
+
+    }
+
+    private SearchObjectsOutput searchObjects(final IndexingStorage idx, final kbasesearchengine.PostProcessing pp) throws Exception {
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: \> 100 chars

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 33]
```diff
@@ -277,5 +285,136 @@ private void searchTypesCheckMatchFilter(
...
+                .withSkipKeys(1L)
+                .withSkipInfo(1L)
+                .withSkipData(1L);
+
+        final SearchObjectsOutput res1 = searchObjects(idx, option1);
+
+        final Map<String, List<String>> highlightRes = new HashMap<>();
+        highlightRes.put("field", highlight);
+        assertThat("did not find objects", res1.getObjects().size() == 1, is(true));
+        assertThat("did not get right highlight", res1.getObjects().get(0).getHighlight(), is(highlightRes));
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: \> 100 chars

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 34]
```diff
@@ -277,5 +285,136 @@ private void searchTypesCheckMatchFilter(
...
+
+        //highlight on and ids on
+        kbasesearchengine.PostProcessing option2 = new kbasesearchengine.PostProcessing()
+                .withIncludeHighlight(1L)
+                .withIdsOnly(1L);
+
+        final SearchObjectsOutput res2 = searchObjects(idx, option2);
+
+        assertThat("did not find objects", res2.getObjects().size() == 1, is(true));
+        assertThat("did not get right highlight", res2.getObjects().get(0).getHighlight(), is(Collections.emptyMap()));
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: \> 100 chars

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java [hunk 35]
```diff
@@ -277,5 +285,136 @@ private void searchTypesCheckMatchFilter(
...
+
+
+        //highlight off and ids off
+        kbasesearchengine.PostProcessing option3 = new kbasesearchengine.PostProcessing()
+                .withIdsOnly(1L);
+
+        final SearchObjectsOutput res3 = searchObjects(idx, option3);
+
+        assertThat("did not find objects", res3.getObjects().size() == 1, is(true));
+        assertThat("did not get right highlight", res3.getObjects().get(0).getHighlight(), is(Collections.emptyMap()));
```
Comment (MrCreosote) [inline test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: \> 100 chars

File: test/src/kbasesearchengine/test/main/SearchMethodsTest.java (hunk 32 above)
Comment (tiramisu24) [reply test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: how do you eyeball this?!

Comment (MrCreosote) [reply test/src/kbasesearchengine/test/main/SearchMethodsTest.java]: If it looks too long, I C&P into an editor and check

**Pull request merged**